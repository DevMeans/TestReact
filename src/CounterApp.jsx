import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    /* setCounter(counter+1);*/
    setCounter((c) => c + 1);
  };
  const handledif = () => {
    /* setCounter(counter+1);*/
    setCounter((c) => c - 1);
  };
  const handleReset = () => {
    /* setCounter(counter+1);*/
    setCounter((c) => c =0);
  };

  return (
    <>
      <div>CounterApp</div>
      <h1>{counter}</h1>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handledif}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
CounterApp.defaultProps = {
  value: 0,
};
