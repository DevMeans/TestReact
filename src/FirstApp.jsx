const newMensaje = "Milton";

export const FirstApp = ({title,edad}) => {
  return (
    <>
      {/* <div>FirstApp</div>*/}

      <p>Soy {title}</p>
      <span>{edad}</span>
    </>
  );
};
