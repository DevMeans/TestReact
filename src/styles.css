html,body{
    background-color: #21232a;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}
button{
    background-color: blanchedalmond;
    padding: 10px;
    border-radius: 50%;
    color: black;
    cursor: pointer;
}