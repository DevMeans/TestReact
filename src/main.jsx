import React from "react";
import ReactDom from "react-dom/client";
import {HelloWorldApp} from "./HelloWorldApp"
import {FirstApp} from "./FirstApp"
import {CounterApp} from "./CounterApp"
import './styles.css'
ReactDom.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
   {/*<FirstApp title="Myriam" edad={32} apellido="Taipe"/>*/} 
    <CounterApp value={2} />
  </React.StrictMode>

)