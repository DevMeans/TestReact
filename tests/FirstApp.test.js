import { render } from "@testing-library/react";
import {FirstApp} from '../src/FirstApp'
describe("Pruebas en FirstApp", () => {
  test("Debe hacer macth con el snapshot", () => {
      const title = "Myriam"
      const {container}= render(<FirstApp title={title} />)
       expect(container).toMatchSnapshot();

  });
});
