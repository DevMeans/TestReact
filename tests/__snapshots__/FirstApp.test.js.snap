// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer macth con el snapshot 1`] = `
<div>
  <p>
    Soy 
    Myriam
  </p>
  <p>
    No hay apellido
  </p>
  <span>
    32
  </span>
</div>
`;
